### PR TITLE
Enable Horizontal Migration Test in QAM

### PR DESCRIPTION
**Pull request by lilyeyes**
Enable Horizontal Migration Test in QAM Incidents job groups.
TEAM-9036 - Enable Horizontal Migration Test in QAM

Added 4 test cases for each product.

    create_hdd_gnome_qam_sap
    sap_sles4sap_failing_horizontal_migration_rollback
    sles4sap_horizontal_migration_sapconf
    sles4sap_horizontal_migration_saptune

  

- Related ticket: https://jira.suse.com/browse/TEAM-9036
- Needles: NA
- Verification run:

  15-SP5: https://openqa.suse.de/tests/13764966#dependencies (passed, no LTSS) 
  15-SP4: https://openqa.suse.de/tests/13779136#dependencies (passed, LTSS)
  15-SP3: https://openqa.suse.de/tests/13779304#dependencies (passed, LTSS)
  15-SP2: https://openqa.suse.de/tests/13779317#dependencies (passed, LTSS)
  15-SP1: End of LTSS.
  12-SP5: https://openqa.suse.de/tests/13778201#dependencies (passed, no LTSS)
  qam-create_hdd_sles4sap_gnome: 12-SP5: http://openqa.suse.de/tests/13778972 (passed, VRs for code changes not break other test cases)

### CODE REVIEW

**Review by BillAnastasiadis — APPROVED** [review]
LGTM

File: tests/console/check_os_release.pm [hunk 1]
```diff
@@ -29,6 +29,7 @@ sub run {
         my $product = get_required_var('SLE_PRODUCT');
         $product = 'sles_sap' if (is_sles4sap and is_sle('<15'));
         $product = 'sles' if (is_sles4sap and is_sle('>=15'));
+        $product = 'sles' if (is_sles4sap and check_var('SCC_REGCODE_SLES4SAP', 'invalid_key') and is_sle('=12-SP5'));
```
Comment (mpagot) [inline tests/console/check_os_release.pm:32]:
```suggestion
         $product = 'sles' if ( is_sles4sap and is_sle('>=15')
                                             || (is_sles4sap and check_var('SCC_REGCODE_SLES4SAP', 'invalid_key') and is_sle('=12-SP5')));
```

downside of this suggestion is the longer line 

But maybe we can think about converting the overall block like
```
my $product;
if (is_sles4sap) {
  if  (  $product = 'sles' 
        || (check_var('SCC_REGCODE_SLES4SAP', 'invalid_key') and is_sle('=12-SP5')))  {
       $product = 'sles' 
  } else if (is_sle('<15')) {
        $product = 'sles_sap' ;
   } else if (is_sle('>=15')
                  || ) {
       $product = 'sles' 
  } else {
        $product = get_required_var('SLE_PRODUCT');
 }
else {
   $product = get_required_var('SLE_PRODUCT');
}
```

**Review by alvarocarvajald — APPROVED** [review]

File: tests/console/check_os_release.pm (hunk 1 above)
Comment (lilyeyes) [reply tests/console/check_os_release.pm]: I prefer the 4 lines too it much more clear, so keep the original line.

Comment (lilyeyes) [reply tests/console/check_os_release.pm]: I prefer the 4 lines too it much more clear, so keep the original line.

**Pull request merged**